nextcord/voice_recording/core.py

```diff
@@ -0,0 +1,1174 @@
+    Formats.MP3: export_one_with_ffmpeg,
+    Formats.MP4: export_one_with_ffmpeg,
+    Formats.M4A: export_one_with_ffmpeg,
+    Formats.MKA: export_one_with_ffmpeg,
+    Formats.MKV: export_one_with_ffmpeg,
+    Formats.OGG: export_one_with_ffmpeg,
+    Formats.PCM: export_one_as_PCM,
+    Formats.WAV: export_one_as_WAV,
+}
+
+
+class Silence:
+    """Represents the number of frames where a recording has been silent
+
+    Parameters
+    ----------
+    frames: :class:`int`
+        The number of 20ms frames where this recording was silent
+
+    Attributes
+    ----------
+    frames: :class:`int`
+        The number of 20ms frames where this recording was silent
+    milliseconds: :class:`float`
+        The relative number of milliseconds based on the frames
+    """
+
+    __slots__ = ("frames",)
+
+    def __init__(self, frames: int) -> None:
+        self.frames: int = frames
+
+    @property
+    def milliseconds(self) -> float:
+        return self.frames / (AUDIO_HZ / FRAME_SIZE) * 1000
+
+    def write_to(self, buffer: BufferedIOBase) -> None:
+        """Writes the silence to IO
+
+        Parameters
+        ----------
+        buffer: :class:`BufferedIOBase`
+            The buffer for which to write this silence to.
+        """
+        res, remainder = divmod(self.frames, STRUCT_SIZE)
+
+        # write in a loop as to avoid generating a huge memory buffer
+        if res:
+            for _ in range(res):
+                buffer.write(SILENCE_STRUCT)
+
+        # write the rest
+        if remainder:
+            buffer.write(SILENCE_STRUCT[: remainder * FRAME_SIZE * AUDIO_CHANNELS])
+
+    @classmethod
+    def from_timedelta(cls, silence: int):
+        """Creates a Silence object with the desginated timeframe.
+
+        Parameters
+        ----------
+        silence: :class:`int`
+            The number of Hz to convert to silence
+            E.g. 48,000 for 1 second of silence when the audio frame size is 48,000Hz
+        """
+        half_frames = int(silence / FRAME_SIZE)
+        return None if half_frames <= 0 else cls(half_frames * AUDIO_CHANNELS)
+
+
+class RecordingFilter:
+    """Represents a filter used to restrict the recording.
+    Allowlist overwrites blocklist. If a user is allowlisted and blocklisted, they
+    will not be affected by the blocklist. List will only affect the filter
+    if they are not empty.
+
+    Parameters
+    ----------
+    client: Optional[:class:`RecorderClient`]
+        The recorder client for which to use this filter on.
+        Is optional for creation but not for when used.
+    iterable: Optional[:class:`RecorderClient`]
+        An optional iterable containing pre-existing data to convert to filter.
+
+    Raises
+    ------
+    TypeError
+        The argument passed for any `user` was not :class:`int`, :class:`User` or :class:`Member`
+    """
+
+    __slots__ = ("blocklist", "allowlist", "client")
+
+    def __init__(
+        self,
+        client=None,
```

Comment:
(Has been a long time) But if I remember correctly, I wasn't able to typehint this because the class `client` takes (i.e. RecorderClient) was defined after, while the class also needs to call it in its functions so I can't just move it. If you know how I could typehint it, let me know.